src/clr-angular/forms/select/providers/option-selection.service.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable} from "@angular/core";
+import {Observable, Subject} from "rxjs";
+
+import {ClrOption} from "../option";
+
+@Injectable()
+export class OptionSelectionService {
+    private currentSelection: ClrOption;
```

Comment:
I just realized this is not going to work. Yo are using our `ClrOption` object to track the current selection. But you are going to have to provide an two-way binding for this, and the consumer will need to provide the **value** of the option as the input. As in, the model of the option. The only way to do this with the current structure is going to be to iterate over all the options and find the one whose model is the same, and that wouldn't even be possible for lazy-loaded options.

I know it's all features that haven't been implemented yet, but two-way binding is part of the MVP and we need to anticipate it.

The solution is to make all of select parametrized by the consumer's model type, and store that type for the current selection. `ClrSelect<T>`, `ClrOption<T>`, `OptionSelectionService<T>`, and then `private currentSelection: T;`.

Reply:
Fixed